### ISSUE TITLE
Flag svn-cache-credentials don't work

### ISSUE THREAD

**Issue by kurkowskik** [composer/composer#4331]
I don't want to store credentials used by composer, but after composer update (exampe config below) the credentials are saved to c:\Users\user\AppData\Roaming\Subversion\auth\svn.simple\ 

example config - composer.json:
```
{
  "repositories":
  [
    {
      "type": "svn",
      "url": "https://svn.example.com/vendor/product/",
      "svn-cache-credentials": false
    }
  ],
  "require": 
  {
    "vendor/product": "9.9.9"
  },
  "config":
  {
    "http-basic":
    {
      "svn.example.com":
      {
        "username": "userhere",
        "password": "passhere"
      }
    }
  }
}
```

I have used debug option (-vvv) and the output seems ok (flag --no-auth-cache was added), so i have used process monitor to debug how composer is calling svn (below). After that analysis I have discovered that flag "--no-auth-cache" should also be used while checkout.

```
svn  info --xml
svn  info --xml
svn  ls --verbose --non-interactive  --no-auth-cache --username "userhere" --password "passhere"  "https://svn.example.com/vendorname/productname/trunk"
svn  ls --verbose --non-interactive  --no-auth-cache --username "userhere" --password "passhere"  "https://svn.example.com/vendorname/productname/branches"
svn  ls --verbose --non-interactive  --no-auth-cache --username "userhere" --password "passhere"  "https://svn.example.com/vendorname/productname/tags"
svn  co --non-interactive  --username "userhere" --password "passhere"  "https://svn.example.com/vendorname/productname/tags/9.9.9/@999" "E:\test\vendor/vendorname/productname"
```

correct checkout command:
```
svn  co --non-interactive --no-auth-cache  --username "userhere" --password "passhere"  "https://svn.example.com/vendorname/productname/tags/9.9.9/@999" "E:\test\vendor/vendorname/productname"
```

**Comment by stof** [comment]
Can you submit a pull request fixing the issue ?

**Comment by kurkowskik** [comment]
I did not analyze the source code, only the way as the composer communicates with svn.exe (using external process monitor). Therefore, it will be difficult for me to quickly introduce the fix.

**Comment by stof** [comment]
@Seldaek it looks like the SvnDownloader does not know about the auth cache: https://github.com/composer/composer/blob/c83650f299cfff1049cd61ea72ee5345bd4f92d3/src/Composer/Downloader/SvnDownloader.php#L87 vs https://github.com/composer/composer/blob/c83650f299cfff1049cd61ea72ee5345bd4f92d3/src/Composer/Repository/Vcs/SvnDriver.php#L316-L317
This is why the only the commands run by the driver are skipping the cache.

**Comment by kurkowskik** [comment]
any news? btw it's a bug not feature.

**Comment by Seldaek** [comment]
Fixed by #4700
